test/onnx/test_pytorch_onnx_caffe2_quantized.py

```diff
@@ -34,7 +34,13 @@ def generic_test(self, model, sample_inputs, input_names=None, decimal=3):
         f.seek(0)
         onnx_model = onnx.load(f)
         caffe_res = c2.run_model(onnx_model, dict(zip(input_names, sample_inputs)))[0]
-        np.testing.assert_almost_equal(output.detach().numpy(), caffe_res, decimal=decimal)
+        output_diff = np.absolute(np.squeeze(output.detach().numpy()) - caffe_res)
+        max_diff = np.amax(output_diff)
+
+        # Permute pytorch output to NHWC
+        # This check had to be changed to account for changes in
+        # qnnpack's requant logic.
+        np.testing.assert_(max_diff <= 1, "Maximum absolute difference must be less than 1")
```

Comment:
I think we shouldn't be relaxing the check for all tests. Currently the requant changes will affect only conv and linear ops. For other ops we should still match with C2.

Reply:
This function is used in multiple places. Maybe we can add a flag for relaxed test. However, the problem is this test is flaky. So imagine you add a new test with conv/linear ops in it and dont set relaxed_compare=True. The test may pass then but it may randomly break. So then you will have this test broken in master. Since this test is data-dependent there is some risk of that. If we are willing to live with it and fix as things break, I am ok with it, but I want to point this out.

Reply:
Yeah I think adding a flag is fine. Currently it only needs to be set for the `test_small_model` case as the conv and linear tests seem to have their own checking. Please add a comment explaining when/why the flag needs to be set.

Reply:
@supriyar Done.